core/src/main/java/hudson/scheduler/CronTab.java

```diff
@@ -103,6 +123,11 @@ private void set(String format, int line, Hash hash) throws ANTLRException {
      * Returns true if the given calendar matches
      */
     boolean check(Calendar cal) {
```

Comment:
IMO the method should not modify the external object. Such behavior is not obvious for users. If it's required due to any reason, the method should be documented with the explicit description of the behavior.